devices/lavender/initramfs/init

```diff
@@ -1,33 +1,43 @@
-#!/bin/sh
+#!/sbin/busybox sh
 
-export PATH=/bin
-
-USERDATA=/dev/block/mmcblk0p66
+USERDATA=/dev/mmcblk0p66
 FSROOT=/mnt/userdata
 NEWROOT=$FSROOT/gnu
 NEWINIT=/sbin/init
 
 # write log to kmsg
 log() {
-	echo new_era: $@ > /dev/kmsg
+	echo "new_era: $@" > /dev/kmsg
 }
 
 die() {
 	log FATAL: $@
 	exit 1
 }
 
-# set up psuedo-filesystems
-mount -t devtmpfs none /dev
+# set up psuedo-filesystems, /dev is already present
+/sbin/busybox mkdir /sys /proc || die Failed to make directories /sys and /proc
+
+# mount the pseudofs(s)
+/sbin/busybox mount -t proc none /proc || die Failed to mount /proc
+/sbin/busybox mount -t sysfs none /sys || die Failed to mount /sys
+/sbin/busybox mount -t tmpfs none /dev	|| die Failed to mount /dev
+
+# populate /dev
+/sbin/busybox mdev -s || die Failed to populate /dev
 
 log Preinit started!
 log Trying to mount $USERDATA on $FSROOT...
-mkdir -p $FSROOT
-mount -t ext4 $USERDATA $FSROOT || die Failed to mount $USERDATA
+/sbin/busybox mkdir -p $FSROOT	|| die Failed to make the directory $FSROOT
+ 
+/sbin/busybox mount $USERDATA $FSROOT || die Failed to mount $USERDATA
 
 log Setting up $NEWROOT as mountpoint...
-mount -o bind,ro $NEWROOT $NEWROOT || die Failed to setup $NEWROOT as mountpoint
+/sbin/busybox mount -r --bind $NEWROOT $NEWROOT || die Failed to setup $NEWROOT as mountpoint
 
 log Cleaning up mounts, switching root to $NEWROOT, and launching $NEWINIT...
-mount --move /dev $NEWROOT/dev
-exec switch_root $NEWROOT $NEWINIT
+/sbin/busybox mount --move /dev $NEWROOT/dev || die Failed to move /dev
+/sbin/busybox mount --move /sys $NEWROOT/sys || die Failed to move /sys
+/sbin/busybox mount --move /proc $NEWROOT/proc || die Failed to move /proc
```

Comment:
Why are all of the commands prefixed with `/sbin/busybox`? Shouldn't those commands be symlinked to that binary already?

Reply:
In my experience, I didn't have access to the UART logs. So I kept the setup as simple as possible. I could enable the symlinking if that is what we aim for.

Reply:
Well, we do not actually care about the initramfs environment as we're doing a `switch_root` soon, so whatever works here will be okay.  When your script starts to get longer, however, the symlinks would allow you to get rid of the busybox prefix and make things easier.